include/oneapi/mkl/detail/export.hpp

```diff
@@ -22,7 +22,7 @@
 
 #include "oneapi/mkl/detail/config.hpp"
 
-#if !defined(BUILD_SHARED_LIBS) || !defined(_WIN64)
```

Comment:
We rely here on the [standard CMake variable](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html) I don't think we need to introduce something new for this project.

Reply:
Thank you for your feedback @mkrainiuk 
`BUILD_SHARED_LIBS` is a standard variable in CMake, but, in this case, it was converted to be used inside C++ project. I don't think users should rely upon any macro defined in oneMath a part from the version macro.

Reply:
To clarify my comment, I believe users should avoid relying on any macros defined within this library for internal usage. While `BUILD_SHARED_LIBS` is a standard option in CMake, converting it into a C++ macro can lead to issues if users wish to use this macro in their own projects. Additionally, it’s not a C++ standard macro. Therefore, I think it would be safer to make this macro project-specific by adding the prefix, as it has been done with other macros in this PR